nautobot_ssot/metrics.py

```diff
@@ -22,25 +22,37 @@ def metric_ssot_jobs():
     for job in Job.objects.filter(slug__icontains="ssot"):
         last_job_sync = Sync.objects.filter(job_result__job_model_id=job.id).last()
         if last_job_sync:
-            ssot_job_durations.add_metric(
-                labels=["source_load_time", job.slug],
-                value=last_job_sync.source_load_time.microseconds,
-            )
-
-            ssot_job_durations.add_metric(
-                labels=["target_load_time", job.slug],
-                value=last_job_sync.target_load_time.microseconds,
-            )
-
-            ssot_job_durations.add_metric(
-                labels=["diff_time", job.slug],
-                value=last_job_sync.diff_time.microseconds,
-            )
-
-            ssot_job_durations.add_metric(
-                labels=["total_sync_time", job.slug],
-                value=last_job_sync.sync_time.microseconds,
-            )
+            if last_job_sync.source_load_time:
+                ssot_job_durations.add_metric(
+                    labels=["source_load_time", job.slug],
+                    value=(last_job_sync.source_load_time.seconds * 100000)
+                    + last_job_sync.source_load_time.microseconds,
+                )
+
+            if last_job_sync.target_load_time:
+                ssot_job_durations.add_metric(
+                    labels=["target_load_time", job.slug],
+                    value=(last_job_sync.target_load_time.seconds * 1000000)
+                    + last_job_sync.target_load_time.microseconds,
+                )
+
+            if last_job_sync.diff_time:
+                ssot_job_durations.add_metric(
+                    labels=["diff_time", job.slug],
+                    value=(last_job_sync.diff_time.seconds * 1000000) + last_job_sync.diff_time.microseconds,
```

Comment:
I thought the `microseconds` was including the `seconds`... so, this means that you need adding both metrics to get the final one?

Reply:
I thought so too and that's why I originally used just the microseconds attribute. When I got to testing though the conversion didn't match up until I included the seconds and multiplier. It became really evident with the diff_time:

```
>>> sync.diff_time
datetime.timedelta(seconds=1, microseconds=69464)
```

Reply:
This has been changed back to seconds but including microseconds to get a more precise measurement.